command/core_wrapper.go

```diff
@@ -40,9 +39,10 @@ func (c *CoreWrapper) PluginRequirements() (plugingetter.Requirements, hcl.Diagn
 // ConfiguredArtifactMetadataPublisher returns a configured image bucket that can be used for publishing
 // build image artifacts to a configured Packer Registry destination.
 func (c *CoreWrapper) ConfiguredArtifactMetadataPublisher() (*packerregistry.Bucket, hcl.Diagnostics) {
-	// JSON can only configure its Bucket through Env. variables so if not in PAR mode
-	// we don't really care if bucket is nil or set to a bunch of zero values.
-	if !env.InPARMode() {
+	bucket := c.Core.GetRegistryBucket()
+
+	// If at this point the bucket is nil, it means PAR is not enabled
+	if bucket == nil {
```

Comment:
I know this is mostly done in the `embed-corebuild-publisher` branch, but this core wrapper is here to make the JSON core sort of match the API of the HCL2 core, I'd not put anything else in here.

Reply:
If I remove it the HCP PAR stops working for JSON. I'll leave any change to @nywilken because removing it, if it's possible, I believe it will change a bit of his implemented logic. 
This is mostly a bucket getter and validation